rj_gameplay/rj_gameplay/basic_play_selector.py

```diff
@@ -1,89 +1,271 @@
+                    return BallPos.THEIR_BALL
+        # TODO(1578): Re-implement actual previous logic here.
+
+        return BallPos.FREE_BALL
+
+    @staticmethod
+    def __calc_pileup(
+        world_state: stp.rc.WorldState, game_info: stp.rc.GameInfo
+    ) -> bool:
+        """Computes whether there is a pileup or not.
+        :param world_state:
+        :param game_info:
+        :return: The current BallPos.
+        """
+        our_near_bots = 0
+        their_near_bots = 0
+        for their_bot in world_state.their_robots:
+            if (
+                np.linalg.norm(
+                    np.array(world_state.ball.pos) - np.array(their_bot.pose[0:2])
+                )
+                < MIN_NEAR_BALL_DIST
+            ):
+                their_near_bots += 1
+        for our_bot in world_state.our_robots:
+            if (
+                np.linalg.norm(
+                    np.array(world_state.ball.pos) - np.array(our_bot.pose[0:2])
+                )
+                < MIN_NEAR_BALL_DIST
+            ):
+                our_near_bots += 1
+
+        if (
+            our_near_bots > 0
+            and their_near_bots > 0
+            and our_near_bots + their_near_bots >= 3
+        ):
+            return True
+        # TODO(1578): Re-implement all the logic here.
+
+        return False
+
+
+class BasicPlaySelector(situation.IPlaySelector):
+    """Play selector that returns a play and situation based on world state
+    (see select method)
     """
 
     def __init__(self):
-        self.analyzer = analyzer.Analyzer()
         self.curr_situation = None
         self.curr_play = None
 
     def select(
-        self, world_state: rc.WorldState
+        self, world_state: stp.rc.WorldState
     ) -> Tuple[Optional[situation.ISituation], stp.play.IPlay]:
+        """Returns the best situation and play
+        for the current world state based on a hardcoded
+        decision tree.
+        :param world_state: The current state of the world.
+        :param game_info: The information about the state of the game.
+        :return: The best situation for the current world state.
+        """
+        game_info = world_state.game_info
+        heuristics = HeuristicInformation(world_state, game_info)
 
+        # TODO: add new plays here as created,
+        #  for now will be commented out
+        """if game_info.state == stp.rc.GameState.STOP:
+            return situations.Stop()
+        elif game_info.is_restart():
+            return self.__analyze_restart(world_state, heuristics)"""
         if world_state.game_info is None and self.curr_play is None:
-            return (self.curr_situation, basic_defense.BasicDefense())
+            return (situations.BasicDefense, basic_defense.BasicDefense())
+        elif (
+            heuristics.ball_pos == BallPos.OUR_BALL
+            or heuristics.ball_pos == BallPos.FREE_BALL
+        ):
+            return (situations.BasicOffense, basic_offense.BasicOffense())
```

Comment:
Shouldn't there be a scramble situation or something similar for a free ball? We should also probably change the situation depending on side of in some cases. Like we would probably act differently to a free ball close to our goal, compared to a free ball on their side.